client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx

```diff
@@ -0,0 +1,96 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CompactCard from 'components/card/compact';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isFreeJetpackPlan } from 'lib/products-values';
+import { isJetpackSite } from 'state/sites/selectors';
+
+class DisconnectSurvey extends Component {
+	state = {
+		reasonSelected: 'onlyNeedFree',
+		renderInitial: true,
+	};
+
+	renderFollowUp() {
+		// placeholder
+		return <Card className="disconnect-site__question">{ 'follow-up' }</Card>;
+	}
+
+	logSelection = option => {
+		this.setState( {
+			reasonSelected: option.value,
+			renderInitial: false,
+		} );
+	};
+
+	getOptions() {
+		const { translate, site } = this.props;
+
+		const options = [
+			{ value: 'tooHard', label: translate( 'It was too hard to configure Jetpack' ) },
+			{ value: 'didNotInclude', label: translate( 'This plan didn’t include what I needed' ) },
+		];
+
+		if ( ! isFreeJetpackPlan( site.plan ) ) {
+			options.push( { value: 'onlyNeedFree', label: translate( 'This plan is too expensive' ) } );
+		}
+		return options;
```

Comment:
There's a significant amount of abstraction and indirection here and I wonder how much value it provides.

What I mean is something like this:

```jsx
renderQuestions() {
  const { site, translate } = this.props;
  return (
    <CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
      { translate( 'It was too hard to configure Jetpack' ) }
    </CompactCard>
    <CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
      { translate( 'This plan didn’t include what I needed' ) }
    </CompactCard>
    { ! isFreeJetpackPlan( site.plan ) && (
      <CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
        { translate( 'It was too hard to configure Jetpack' ) }
      </CompactCard>
    }
  );
}
```

Just a thought and absolutely not a change that you need to incorporate 🙂

Reply:
Yep, I agree that componentization is the kind of modularization we want in Calypso.